fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/AccountNumberGenerator.java

```diff
@@ -101,10 +111,9 @@ private String generateAccountNumber(Map<String, String> propertyMap, AccountNum
             if (customLength.getValue() != null) {
                 accountMaxLength = customLength.getValue().intValue();
             }
-
         }
 
-        String accountNumber = StringUtils.leftPad(propertyMap.get(ID), accountMaxLength, '0');
+        accountNumber = StringUtils.leftPad(accountNumber, accountMaxLength, '0');
```

Comment:
Why do we need to initialise this again? It was initialise already in line 103...